lib/amazon-pricing/aws-price-list.rb

```diff
@@ -56,8 +57,22 @@ def get_instance_type(region_name, api_name)
     end
 
     def self.fetch_url(url)
-      uri = URI.parse(url)
-      page = Net::HTTP.get_response(uri)
+      # AWS appears to have started throttling URL accesses, so we now have to retry
+      retry_count = 1
+      while retry_count <= RETRY_LIMIT do
```

Comment:
I am pretty sure you can use Ruby 'retry' keyword to make this more readable, but not a blocker